### ISSUE TITLE
Text output is hard to read

### ISSUE THREAD

**Issue by mhidas** [ioos/compliance-checker#578]
I'm looking at the new text output format from version 4, and often find it hard to read. Individual reasoning messages blend into a column of solid text, and sometimes the names do too, so it takes extra concentration to see where each new message begins. If a file has lots of errors it can look something like this:
```
                                     Errors                                     
--------------------------------------------------------------------------------
Name                                      Reasoning
§3.1 Variable WDIR's units are            units for variable WDIR must be
appropriate for the standard_name         convertible to degree currently they are
wind_to_direction:                        degrees (clockwise from true north)
§3.1 Variable WDIR_2's units are          units for variable WDIR_2 must be
appropriate for the standard_name         convertible to degree currently they are
wind_to_direction:                        degrees (clockwise from true north)
§3.5 AIRT_2_quality_control is a valid    AIRT_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 AIRT_quality_control is a valid      AIRT_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 ATMP_2_quality_control is a valid    ATMP_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 ATMP_quality_control is a valid      ATMP_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 CNDC_2_quality_control is a valid    CNDC_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 CNDC_quality_control is a valid      CNDC_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 LATITUDE_quality_control is a valid  LATITUDE_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 LONGITUDE_quality_control is a       LONGITUDE_quality_control does not
valid flags variable:                     define either flag_masks or flag_values
§3.5 LW_2_quality_control is a valid      LW_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 LW_quality_control is a valid flags  LW_quality_control does not define
variable:                                 either flag_masks or flag_values
§3.5 PL_CMP2_quality_control is a valid   PL_CMP2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 PSAL_2_quality_control is a valid    PSAL_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 PSAL_quality_control is a valid      PSAL_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 RAIN_AMOUNT_2_quality_control is a   RAIN_AMOUNT_2_quality_control does not
valid flags variable:                     define either flag_masks or flag_values
§3.5 RAIN_AMOUNT_quality_control is a     RAIN_AMOUNT_quality_control does not
valid flags variable:                     define either flag_masks or flag_values
§3.5 RELH_2_quality_control is a valid    RELH_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 RELH_quality_control is a valid      RELH_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 SW_2_quality_control is a valid      SW_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 SW_quality_control is a valid flags  SW_quality_control does not define
variable:                                 either flag_masks or flag_values
§3.5 TEMP_2_quality_control is a valid    TEMP_2_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 TEMP_quality_control is a valid      TEMP_quality_control does not define
flags variable:                           either flag_masks or flag_values
§3.5 TIME_quality_control is a valid      TIME_quality_control does not define
flags variable:                           either flag_masks or flag_values

etc...
```
It would be nice to tweak the output formatting so that individual message are better separated.

I also think there is too much redundancy between a given error's "name" and the "reasoning" message. Do we really need to have the offending variable's name mentioned in both? In fact, I don't know why we even need two columns. We're just listing corrective actions, which could all be detailed in the reasoning messages.

**Comment by mhidas** [comment]
The simplest solution I can suggest is to separate each message with a blank line.

If we also get rid of the "name" column, we can have a simple bullet list of reasoning messages, like this:
```
                                     Errors                                     
--------------------------------------------------------------------------------
* units for variable WDIR must be convertible to degree currently they are
  degrees (clockwise from true north)

* units for variable WDIR_2 must be convertible to degree currently they are
  degrees (clockwise from true north)

* AIRT_quality_control does not define either flag_masks or flag_values

* ATMP_quality_control does not define either flag_masks or flag_values

* CNDC_quality_control does not define either flag_masks or flag_values

* LATITUDE_quality_control does not define either flag_masks or flag_values

* LONGITUDE_quality_control does not define either flag_masks or flag_values
```

Or if you still want to make use of the name field (minus variable names) to group similar error messages together and (at least for CF) provide a reference to the relevant section in the convention document. Maybe something like this?
```
                                     Errors                                     
--------------------------------------------------------------------------------
§3.1 Variable units are appropriate for the standard_name
  * units for variable WDIR must be convertible to degree currently they are
    degrees (clockwise from true north)

  * units for variable WDIR_2 must be convertible to degree currently they are
    degrees (clockwise from true north)

§3.5 Flags
  * AIRT_quality_control does not define either flag_masks or flag_values

  * ATMP_quality_control does not define either flag_masks or flag_values

  * CNDC_quality_control does not define either flag_masks or flag_values

  * LATITUDE_quality_control does not define either flag_masks or flag_values

  * LONGITUDE_quality_control does not define either flag_masks or flag_values
```

**Comment by Bobfrat** [comment]
@mhidas thanks for the feedback. I like the 2nd option better so we can keep the relevant CF section in the report.

**Comment by mhidas** [comment]
@benjwadams Thanks for the info.
If you like my suggested format, do you still need tables at all?
If the options to output LaTeX tables is important, I guess you could still use `tabulate` to create single-column tables. But maybe in that case this output format could be called "table" (or "latex") rather than "text"?

For a plain text output format, I would prefer the simplest possible output.

Personally I don't even think it's worth wrapping the output at all. Each reasoning message could just be a single continuous line of text. That way I can e.g. view it in a terminal of whatever width, and let my paging tool take care of the wrapping, or paste it into an email. Having it wrapped is ok, as long as it's just one column and the width is closer to 80 characters, rather than the current 40. (I can also generate my own text output from the json if I have to.)

**Comment by benjwadams** [comment]
Should have been addressed by #616, closing.

**Comment by mhidas** [comment]
Looks much better. Thank you!
